industry_benchmarks/utils/data_gathering.py

```diff
@@ -0,0 +1,982 @@
+
+    Returns
+    -------
+    The name of the transformation as a tuple of (phase, ligand_a name, ligand_b name)
+    """
+    # grab the gufe key of the chemical systems used in the inputs
+    unit_result = list(result["unit_results"].values())[0]
+    state_a_key = unit_result["inputs"]["stateA"][":gufe-key:"]
+    mapping_key = unit_result["inputs"]["ligandmapping"][":gufe-key:"]
+    # work out which system this is in the alchemical network
+    system_look_up = dict((str(node.key), node) for node in alchemical_network.nodes)
+    mapping_look_up = dict((str(edge.mapping.key), edge.mapping) for edge in alchemical_network.edges)
+    # build the transform
+    if any([isinstance(comp, gufe.ProteinComponent) for comp in system_look_up[state_a_key].components.values()]):
+        phase = "complex"
+    else:
+        phase = "solvent"
+
+    ligmap = mapping_look_up[mapping_key]
+    return phase, ligmap.componentA.name, ligmap.componentB.name
+
+def check_network_is_connected(results_data: dict[tuple[str, str, str], list[tuple[unit.Quantity, unit.Quantity, pathlib.Path]]], alchemical_network: gufe.AlchemicalNetwork, name_mapping: None | dict[str, str] = None) -> bool:
+    """
+    Build a network from the results and check the network is connected.
+
+    This requires that both the complex and solvent phases have 3 repeats.
+
+    """
+    from networkx.exception import NetworkXPointlessConcept
+
+    # reverse the name mapping
+    if name_mapping is not None:
+        reverse_name_mapping = dict(
+            (v, k) for k, v in name_mapping.items()
+        )
+    else:
+        reverse_name_mapping = None
+
+    # create a new version of the results data with the correct edge names
+    results_by_edge = {}
+    edge_to_ligands = {}
+    for (phase, ligand_a, ligand_b), results in results_data.items():
+        # create the edge string
+        if reverse_name_mapping is not None:
+            edge_name = f"{phase}_{reverse_name_mapping[ligand_a]}_{reverse_name_mapping[ligand_b]}"
+
+        else:
+            edge_name = f"{phase}_{ligand_a}_{ligand_b}"
+
+        results_by_edge[edge_name] = results
+        edge_to_ligands[edge_name] = (ligand_a, ligand_b)
+
+
+    edges = defaultdict(list)
+    # group the transforms by ligands
+    for transform in alchemical_network.edges:
+        if transform.name in results_by_edge and len(results_by_edge[transform.name]) == 3:
+            # get the names of the ligands in this edge
+            ligand_a, ligand_b = edge_to_ligands[transform.name]
+            edges[(ligand_a, ligand_b)].append(transform)
+
+    # extract edges which have both phases completed
+    complete_edges = [t for values in edges.values() if len(values) == 2 for t in values]
+
+    # extract the ligand network and check its connected
+    result_network = gufe.AlchemicalNetwork(edges=complete_edges)
+    ligand_network = extract_ligand_network(result_network)
```

Comment:
This is a perfectly fine strategy. In the future we should see if we can extend the `.graph` method of AlchemicalNetwork objects to offer this kind of convenience. I.e. "get me all the connected subgraphs".

Reply:
Great idea, made an issue on [gufe](https://github.com/OpenFreeEnergy/gufe/issues/383).